modules/candidate_parameters/jsx/ProbandInfo.js

```diff
@@ -97,6 +97,66 @@ var ProbandInfo = React.createClass(
         dob2Required = true;
       }
 
+      var extraParameterFields = [];
+      var extraParameters = this.state.Data.extra_parameters;
+      for (var key2 in extraParameters) {
+        if (extraParameters.hasOwnProperty(key2)) {
```

Comment:
Rather than use `for ... in` and doing this `hasOwnProperty` check you could use `forEach`, which won't iterate inherited properties. 
```extraParameters.forEach(function(extraParam) { ... })``` 

Better yet would be to use `map`. It looks like you are creating a component for each `extraParameter`, i.e. a 1-to-1 mapping.

I'd write it like:
```
var extraParameters = this.state.Data.extra_parameters;
var extraParametersFields = extraParameters.map(function(extraParam) { ... });
```

If you decide to use `forEach` or `map`, you'll need to change `extraParameters[key2]` to `extraParam` in the body of the function. If you use `map`, instead of `push`ing, just return the component.